### DISCUSSION TITLE
Beware of recent BC

### DISCUSSION THREAD

**Discussion by aprat84** [doctrine/doctrine2#11427] [General]
Recently this PR was merged and caused a bunch of BC: https://github.com/doctrine/orm/pull/8391

This is just a notice to the authors, cause this project is supposed to be mature and super cautious about BC, and from a single PR at least 2 BC where created, that I know: https://github.com/doctrine/orm/issues/11097, https://github.com/doctrine/orm/issues/11128

**Comment by beberlei** [comment]
Nobody is perfect, especially in their free time.

**Reply by aprat84** [reply]
Well, #11097 is a side effect, hard to notice.

But when you disallow WITH keyword on fetch joined associations configured with fetch=EAGER, its pretty obvious that it was gona generate BC: before you could do something that now can't.

Don't get me wrong, I don't mind BC, but always on major updates please!
This library is installed more than 100k times a day lately...

**Reply by beberlei** [reply]
Why are you not just staying on 2.x? 

> We plan to maintain Doctrine ORM 2 for at least the next 2 years by providing bug and security fixes. We may also add or deprecate things in 2.x to improve the existing forward-compatbility layer to make the transition to ORM 3 smoother.

**Reply by aprat84** [reply]
These 2 BC were in v2... 🤷🏼‍♂️
